src/components/views/emojipicker/Preview.tsx

```diff
@@ -34,10 +34,10 @@ class Preview extends React.PureComponent<IProps> {
                 </div>
                 <div className="mx_EmojiPicker_preview_text">
                     <div className="mx_EmojiPicker_name mx_EmojiPicker_preview_name">
-                        { label }
+                        { label.length > 28 ? label.slice(0, 28) + '...': label }
                     </div>
                     <div className="mx_EmojiPicker_shortcode">
-                        { shortcode }
+                        { shortcode.length > 28 ? shortcode.slice(0, 32) + '...': shortcode }
```

Comment:
Lets please use ellipsis in CSS as users can change their fonts/size so a fixed number of characters will break elsewhere.

Reply:
It's not working that's why I did that... You can pls check it and see

Reply:
Share what you tried?

Reply:
I am not with my PC currently, I gave the two text, `text-overflow:ellipses` but it's not working because the description container is not relative to the emoji picker panel but rather the entire chat screen... I don't know if I am clear. ... You can try at your side by inspecting the text and include it if it works

Reply:
Ok let me try this and see, Thanks

Reply:
> You'll also need to specify `overflow` and maybe `white-space`

This is not working as well

Reply:
```css
.mx_EmojiPicker_preview_text {
  width: calc(100% - 64px);
}

.mx_EmojiPicker_shortcode {
  overflow: hidden;
  text-overflow: ellipsis;
}
```

The above works for me

Reply:
Thanks, it works but still seem a bit off if you check the red line i placed at the end, also the original ticket opener suggested the description to be in one line not wrapped, but if it's fine by you to be wrapped no problem.

![emoji](https://user-images.githubusercontent.com/38439166/168387703-51d165aa-ff8e-4a3f-b3c8-229c64ad06b4.PNG)

Reply:
> Lets please use ellipsis in CSS as users can change their fonts/size so a fixed number of characters will break elsewhere.

Also concerning this. i tried to make my screen larger to see if my solution will break but it didn't

Reply:
![image](https://user-images.githubusercontent.com/2403652/168391435-aae2d024-5aec-4a9f-8750-b85cfcef5f62.png)

This is with your code, with Settings > Appearance font size 20pt and font `monospace`

Reply:
> Thanks, it works but still seem a bit off if you check the red line i placed at the end, also the original ticket opener suggested the description to be in one line not wrapped, but if it's fine by you to be wrapped no problem.

If you want to limit it to one line just use `white-space: nowrap;`

Reply:
Haa, Thanks. But when i included `white-space: nowrap;`. it overflows as shown in the image below, i also include `text-overflow: ellipses` but it didn't have effect on it

![emoji](https://user-images.githubusercontent.com/38439166/168393318-4ee05bb1-c3b2-41fb-b683-1ca4255eb85d.PNG)

Reply:
![image](https://user-images.githubusercontent.com/2403652/168393918-3730ed66-f90a-428d-a324-2b179c79ebbf.png)
![image](https://user-images.githubusercontent.com/2403652/168393947-53a0890b-1989-42ad-a2e7-c9357eb57a58.png)
![image](https://user-images.githubusercontent.com/2403652/168393957-e5945120-0948-4fa5-aec8-2a9dac8ecc2c.png)
![image](https://user-images.githubusercontent.com/2403652/168393966-0ad475ed-4d65-43ed-b9a6-eca14bf9a952.png)

Reply:
Ok, figured i had to include `overflow: hidden`

Reply:
Thanks very much, let me push the changes